test/e2e/declarative_test_fwk/helper.py

```diff
@@ -201,16 +282,57 @@ def assert_items(self, expectations: dict, state: dict) -> None:
         if not expectations:
             # nothing to assert as there are no expectations
             return
-        if not state:
-            # there are expectations but no given state to validate
+        if not k8s_resource:
+            # there are expectations but no given k8s_resource state to validate
             # following assert will fail and assert introspection will provide useful information for debugging
-            assert expectations == state
+            assert expectations == k8s_resource
 
         for (key, value) in expectations.items():
             # conditions are processed separately
             if key == "conditions":
                 continue
-            assert (key, value) == (key, state.get(key))
+            # tags are processed by custom logic
+            if key == "tags":
+                self.assert_tags(value, k8s_resource.get(key))
+                continue
+            # assert any specific extra fields
+            if self.assert_extra_items_k8s(key, value, k8s_resource):
+                continue
+            assert value == k8s_resource.get(key)
+
+    def assert_extra_items_k8s(self, name: str, value: any, k8s_resource: dict) -> bool:
```

Comment:
nit: The parameters for this method have different names and order to `assert_extra_items_aws`

Reply:
Made them consistent.